recipe/meta.yaml

```diff
@@ -1,92 +1,100 @@
     script: build-core.bat  # [win]
     requirements:
       build:
-        {% if not win %}
-        - {{ compiler('c') }}
-        - {{ compiler('cxx') }}
-        - {{ compiler('fortran') }}
-        {% else %}
-        - {{ n }}toolchain
-        {% endif %}
-
-        - perl  # [not win]
+        - {{ compiler('go14') }}
+
+        - perl     # [not win]
+        - m2-sed   # [win]
+        - m2-grep  # [win]
+
+        - {{ compiler('c') }}              # [unix and cgo]
+        - {{ compiler('m2w64_c') }}        # [win and cgo]
+      run:                                 # [cgo]
+        - {{ compiler('c') }}              # [unix and cgo]
+        - {{ compiler('m2w64_fortran') }}  # [win and cgo]
     test:
       files:
         - hello.go
+        - hello-c.go  # [cgo]
       commands:
+        # Environment checks
         - go env
+        - test $(go env GOROOT) = ${CONDA_PREFIX}/go         # [unix]
+        - cmd /c if x%GOROOT% NEQ x%CONDA_PREFIX%\go exit 1  # [win]
+
+        - test $(go env CGO_ENABLED) -eq {{ cgo }}  # [unix]
+
+        # Test go can actually compile and run a basic program
         - go run hello.go
-        - cmd /c echo x%GOROOT% NEQ x%CONDA_PREFIX%\go        # [win]
-        - cmd /c if x%GOROOT% NEQ x%CONDA_PREFIX%\go exit 1   # [win]
 
-  - name: {{ name|lower }}_{{ conda_goplatform }}
-    script: build-platform.sh   # [not win]
-    script: build-platform.bat  # [win]
+        # Test cgo is working properly
+        - go run hello-c.go    # [cgo]
+
+  - name: {{ name }}_{{ target_platform }}         # [cgo]
+  - name: {{ name }}-nocgo_{{ conda_goplatform }}  # [not cgo]
+    script: build-{{ cgo_var }}-platform.sh   # [unix]
+    script: build-{{ cgo_var }}-platform.bat  # [win]
     requirements:
       build:
         - sed      # [osx]
         - m2-sed   # [win]
-        - m2-file  # [win]
-        - m2-grep  # [win]
-      host:
-        - {{ pin_subpackage("go-core", exact=True) }}
       run:
-        - {{ pin_subpackage("go-core", exact=True) }}
+        - {{ pin_subpackage("go-"+cgo_var, exact=True) }}
     test:
       files:
         - hello.go
+        - hello-c.go
       requires:
-        - posix  # [win]
+        - posix    # [win]
+        - m2-file  # [win]
+        - m2-grep  # [win]
       commands:
         - go env
-        - go build hello.go
+
         # Check that the file types match what is expected for each platform
         ## https://stackoverflow.com/questions/2768608/batch-equivalent-of-bash-backticks#2768660
-        - file hello`go env GOEXE` | grep '{{ conda_gofile }}'  # [unix]
-        - for /f "usebackq tokens=*" %%a in (`go env GOEXE`) do file hello%%a | grep '{{ conda_gofile }}'  # [win]
+        - go build hello.go
+        - file hello`go env GOEXE` | grep '{{ conda_gofile }}'  # [unix and not cgo]
+        - for /f "usebackq tokens=*" %%a in (`go env GOEXE`) do file hello%%a | grep '{{ conda_gofile }}'  # [win and not cgo]
```

Comment:
Checks that the cross-compiling is actually working. We can't really execute then locally, so I am using `file` to check the file signature and compare against the `conda_gofile` variant.